graphql_compiler/tests/test_input_data.py

```diff
@@ -2465,3 +2465,30 @@ def complex_nested_optionals():
         expected_output_metadata=expected_output_metadata,
         expected_input_metadata=expected_input_metadata,
         type_equivalence_hints=None)
+
+
+def recursive_field_type_is_subtype_of_parent_field():
+    """Ensure that traversal is allowed along an edge linked to a supertype of the parent field."""
```

Comment:
nit: `recursion` instead of `traversal` -- a vanilla traversal is always allowed

Reply:
I think you forgot to fix this ^ in this instance -- you fixed the other one though.